From 5c72f5178380d4c1480d6c1ca9bb4322c1efaf70 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABtan=20Andr=C3=A9?= <rvlander@gaetanandre.eu>
Date: Sun, 8 Mar 2020 09:46:41 +0100
Subject: [PATCH 121/210] FIXME: arm64: dts: qcom:
 apq8016-samsung-matissevewifi: add display (v3)

v2: Make reset-gpios GPIO_ACTIVE_LOW
v3: Add &mdss { status = "okay"; };
---
 .../qcom/apq8016-samsung-matissevewifi.dts    | 100 ++++++++++++++++++
 1 file changed, 100 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/apq8016-samsung-matissevewifi.dts b/arch/arm64/boot/dts/qcom/apq8016-samsung-matissevewifi.dts
index f1e2eab9174..7a13001ee85 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-samsung-matissevewifi.dts
+++ b/arch/arm64/boot/dts/qcom/apq8016-samsung-matissevewifi.dts
@@ -48,6 +48,40 @@ home {
 			linux,code = <KEY_HOMEPAGE>;
 		};
 	};
+
+	reg_lcd: regulator-lcd {
+		compatible = "regulator-fixed";
+		regulator-name = "lcd";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+
+		/* FIXME: find why we have to force regulator on */
+		regulator-always-on;
+		regulator-boot-on;
+
+		gpio = <&msmgpio 8 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&lcd_en_default>;
+	};
+
+	reg_lvds: regulator-lvds {
+		compatible = "regulator-fixed";
+		regulator-name = "lvds";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+
+		/* FIXME: see above */
+		regulator-always-on;
+		regulator-boot-on;
+
+		gpio = <&msmgpio 17 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&lvds_en_default>;
+	};
 };
 
 &blsp_i2c1 {
@@ -90,6 +124,39 @@ &blsp1_uart2 {
 	status = "okay";
 };
 
+&dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+
+	panel@0 {
+		compatible = "samsung,tc358764-ltl101al06";
+		reg = <0>;
+
+		lcd-supply = <&reg_lcd>;
+		lvds-supply = <&reg_lvds>;
+		reset-gpios = <&msmgpio 31 GPIO_ACTIVE_LOW>;
+
+		clocks = <&gcc GCC_GP3_CLK>;
+		clock-names = "pwm";
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&dsi0_out>;
+			};
+		};
+	};
+};
+
+&dsi0_out {
+	data-lanes = <0 1 2 3>;
+	remote-endpoint = <&panel_in>;
+};
+
+&mdss {
+	status = "okay";
+};
+
 &pm8916_resin {
 	status = "okay";
 	linux,code = <KEY_VOLUMEDOWN>;
@@ -250,6 +317,39 @@ gpio_keys_default: gpio-keys-default {
 		bias-pull-up;
 	};
 
+	lcd_en_default: lcd-en-default {
+		pins = "gpio8";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	lvds_en_default: lvds-en-default {
+		pins = "gpio17";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	mdss {
+		mdss_default: mdss-default {
+			pins = "gpio31";
+			function = "gpio";
+
+			drive-strength = <8>;
+			bias-disable;
+		};
+		mdss_sleep: mdss-sleep {
+			pins = "gpio31";
+			function = "gpio";
+
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+	};
+
 	muic_int_default: muic-int-default {
 		pins = "gpio12";
 		function = "gpio";
-- 
2.34.1

